### PR TITLE
Fix workflows cycle 

### PR DESCRIPTION
**Pull request by scott45**
- Make the cycle functional
- Clean up workflows that are not being used.
- Use the least amount of steps possible
- Use matrixes

Corresponding issue - https://github.com/GluuFederation/cloud-native-edition/issues/405

### CODE REVIEW

File: .github/workflows/central_code_quality_check.yml [hunk 1]
```diff
@@ -8,14 +8,21 @@ on:
...
+      - 4.3
   workflow_dispatch:
+
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        fetch-depth: [0]
+        java-version: [11]
```
Comment (moabu) [inline .github/workflows/central_code_quality_check.yml]: using a matrix means there will be more then one option provided. Here, that is not true.  Both are static and we won't be changing that.

File: .github/workflows/central_code_quality_check.yml [hunk 2]
```diff
@@ -33,13 +40,13 @@ jobs:
     steps:
       - uses: actions/checkout@v2
         with:
-          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of sonarqube analysis
+          fetch-depth: ${{ matrix.fetch-depth }}  # Shallow clones should be disabled for a better relevancy of sonarqube analysis
```
Comment (moabu) [inline .github/workflows/central_code_quality_check.yml]: Note the above comment.

File: .github/workflows/central_code_quality_check.yml [hunk 3]
```diff
@@ -33,13 +40,13 @@ jobs:
...
 
       - name: Set up JDK 11
         if: contains(env.JVM_PROJECTS, github.repository)
         uses: actions/setup-java@v2.5.0
         with:
-          java-version: '11'
+          java-version: ${{ matrix.java-version }}
```
Comment (moabu) [inline .github/workflows/central_code_quality_check.yml]: Note the above comment.

File: .github/workflows/central_sync_workflows.yml [hunk 4]
```diff
@@ -13,13 +13,16 @@ on:
...
+
     steps:
       - name: Fetching Local Repository
         uses: actions/checkout@master
+        
       - name: Import GPG key
         id: import_gpg
         uses: crazy-max/ghaction-import-gpg@v4
         with:
+          # aadded and saved
```
Comment (moabu) [inline .github/workflows/central_sync_workflows.yml]: whats this comment saying ?

File: .github/workflows/commit-check.yml [hunk 5]
```diff
@@ -20,16 +20,22 @@ jobs:
   check-commit-message:
     name: Check Commit Message
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [14]
+        fetch-depth: [2]
```
Comment (moabu) [inline .github/workflows/commit-check.yml]: using a matrix means there will be more then one option provided. Here, that is not true. Both are static and we won't be changing that.

File: .github/workflows/commit-check.yml [hunk 6]
```diff
@@ -20,16 +20,22 @@ jobs:
...
+
     steps:
       - name: Checkout Project
         uses: actions/checkout@v2
         with:
           # We need to fetch with a depth of 2 for pull_request so we can do HEAD^2
-          fetch-depth: 2
+          fetch-depth: ${{ matrix.fetch-depth }}
```
Comment (moabu) [inline .github/workflows/commit-check.yml]: Note the above comment.

File: .github/workflows/commit-check.yml [hunk 7]
```diff
@@ -20,16 +20,22 @@ jobs:
...
 
       - uses: actions/setup-node@v2.5.1
         with:
-          node-version: 14
+          node-version: ${{ matrix.node-version }}
```
Comment (moabu) [inline .github/workflows/commit-check.yml]: Note the above comment.

File: .github/workflows/issues.yaml [hunk 8]
```diff
@@ -0,0 +1,35 @@
+name: Label and Comment on issues
```
Comment (moabu) [inline .github/workflows/issues.yaml]: Remove this file

File: .github/workflows/microk8s.yml [hunk 9]
```diff
@@ -6,22 +6,26 @@ on:
...
     runs-on: ubuntu-latest
     strategy:
       max-parallel: 6
       matrix:
         python-version: [3.7]
-        gluu-version: ['"4.3.0_dev"']
+        gluu-version: ['"4.3.2_dev"']
         test-cases: ["install", "helm-install"]
-        # add '"pgsql" when supported
-        persistence-backends: ['"ldap"','"couchbase"','"sql"']
```
Comment (moabu) [inline .github/workflows/microk8s.yml:21]: Why are we removing couchbase ?

**Review by moabu — CHANGES_REQUESTED** [review]
Please go over my comments. Also couchbase cannot be removed. I think you should install it separately using helm and then connect it inside the workflow to test out the setup with CB.  Have you thought about using IaC

File: .github/workflows/central_code_quality_check.yml (hunk 1 above)
Comment (scott45) [reply .github/workflows/central_code_quality_check.yml]: Thanks, done reverting.

File: .github/workflows/central_code_quality_check.yml (hunk 2 above)
Comment (scott45) [reply .github/workflows/central_code_quality_check.yml]: Reverted

File: .github/workflows/central_code_quality_check.yml (hunk 3 above)
Comment (scott45) [reply .github/workflows/central_code_quality_check.yml]: Reverted

File: .github/workflows/central_sync_workflows.yml (hunk 4 above)
Comment (scott45) [reply .github/workflows/central_sync_workflows.yml]: Removed this. 
I was trying to pass the central sync workflow. I got rid of all the added changes but hadn't caught this

File: .github/workflows/commit-check.yml (hunk 5 above)
Comment (scott45) [reply .github/workflows/commit-check.yml]: Reverted, thanks

File: .github/workflows/commit-check.yml (hunk 6 above)
Comment (scott45) [reply .github/workflows/commit-check.yml]: Reverted

File: .github/workflows/commit-check.yml (hunk 7 above)
Comment (scott45) [reply .github/workflows/commit-check.yml]: Reverted

File: .github/workflows/issues.yaml (hunk 8 above)
Comment (scott45) [reply .github/workflows/issues.yaml]: Done.

File: .github/workflows/microk8s.yml (hunk 9 above)
Comment (scott45) [reply .github/workflows/microk8s.yml:21]: Added it back. 

`Also couchbase cannot be removed. I think you should install it separately using helm and then connect it inside the workflow to test out the setup with CB` 

Used the above approach and it's working. 

`Have you thought about using IaC`

I didn't understand pretty well Sir how to use IAC in GitHub action workflows. But looking currently looking that up

File: .github/workflows/couchbase_microk8s.yml [hunk 10]
```diff
@@ -0,0 +1,217 @@
+name: couchbase
```
Comment (moabu) [inline .github/workflows/couchbase_microk8s.yml:1]: is this working ? and why did you break it off into separate workflows ?

**Review by moabu — CHANGES_REQUESTED** [review]
There should be one microk8s workflow. 
 As for my comment on CB, you can use helm to install CB

**Review by moabu — APPROVED** [review]

**Comment by moabu** [comment]
im merging due to long thread on this. The issue associated still needs work. All microk8s jobs need to be in one workflow

**Pull request merged**